src/components/MultiDropdown/MultiDropdown.tsx

```diff
@@ -1,75 +1,67 @@
-  value: Option[];
+  value: MealType[];
   /** Callback, вызываемый при выборе варианта */
-  onChange: (value: Option[]) => void;
-  /** Заблокирован ли дропдаун */
-  disabled?: boolean;
+  onChange: (value: MealType[]) => void;
   /** Преобразовать выбранные значения в строку. Отображается в дропдауне в качестве выбранного значения */
-  pluralizeOptions: (value: Option[]) => string;
+  pluralizeOptions: (value: MealType[]) => string;
 };
 
-export const MultiDropdown: React.FC<MultiDropdownProps> = ({
-  options,
-  value,
-  onChange,
-  disabled,
-  pluralizeOptions,
-}) => {
-  const [expanded, setExpanded] = React.useState(false);
+export const MultiDropdown: React.FC<MultiDropdownProps> = React.memo(
+  ({ mealTypes, value, onChange, pluralizeOptions }) => {
+    const [expanded, setExpanded] = React.useState(false);
 
-  React.useEffect(() => {
-    return () => setExpanded(false);
-  }, [disabled]);
+    // Функция для эффекта, закрывающего дропдаун при клике куда-то в сторону.
+    const handleSideClick = React.useCallback((event: MouseEvent) => {
+      if (!(event.target as HTMLElement).closest('#multidropdown'))
+        setExpanded(false);
+    }, []);
 
-  const ifSelected = (opt: Option): string => {
-    return value.some((e) => e.key === opt.key) ? 'selected' : 'not-selected';
-  };
+    React.useEffect(() => {
+      if (expanded) document.addEventListener('click', handleSideClick);
+      else document.removeEventListener('click', handleSideClick);
+    }, [expanded]);
 
-  const handleChange = (opt: Option): void => {
-    if (ifSelected(opt) === 'selected') {
-      onChange(value.filter((e) => e.key !== opt.key));
-    } else {
-      onChange(value.concat(opt));
-    }
-  };
+    const ifSelected = (mt: MealType): string => {
+      return value.some((e) => e.key === mt.key)
+        ? 'multiDropdown__selected'
+        : 'multiDropdown__not-selected';
+    };
 
-  return (
-    <div className={styles.multiDropdown}>
-      <div
-        className={
-          disabled
-            ? styles.multiDropdown__header_disabled
-            : styles.multiDropdown__header
-        }
-        onClick={() => setExpanded(disabled ? false : !expanded)}
-      >
-        {pluralizeOptions(value)}
+    const handleChange = (mt: MealType): void => {
+      if (ifSelected(mt) === 'multiDropdown__selected')
+        onChange(value.filter((e) => e.key !== mt.key));
+      else onChange(value.concat(mt));
+    };
+
+    return (
+      <div className={styles.multiDropdown} id="multidropdown">
+        <div
+          className={styles.multiDropdown__header}
+          onClick={() => setExpanded(!expanded)}
+        >
+          {pluralizeOptions(value) || 'Select Dish Categories'}
+        </div>
+        {expanded &&
+          mealTypes.map((mt: MealType) => (
+            <div
+              key={mt.key}
+              className={styles[ifSelected(mt)]}
+              onClick={() => handleChange(mt)}
```

Comment:
для красоты предлагаю немного переделать handleChange на вот такое:
`handleChange = (mt) => (opt: Option) => {...}`
тогда здесь тебе нужно будет написать только:
```suggestion
              onClick={handleChange(mt)}
```